osquery/tables/system/windows/signature.cpp

```diff
@@ -0,0 +1,238 @@
+/*
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#include <codecvt>
+#include <map>
+#include <string>
+
+// clang-format off
+#define _WIN32_DCOM
+#define WIN32_LEAN_AND_MEAN
+
+#include <Windows.h>
+#include <psapi.h>
+#include <stdlib.h>
+#include <tlhelp32.h>
+#include <wincrypt.h>
+#include <Softpub.h>
+// clang-format on
+
+#include <osquery/core.h>
+#include <osquery/core/conversions.h>
+#include <osquery/filesystem.h>
+#include <osquery/logger.h>
+#include <osquery/sql.h>
+#include <osquery/tables.h>
+
+namespace osquery {
+struct SignatureInformation final {
+  enum class Result { Valid, Trusted, Invalid, Missing, Distrusted, Untrusted };
+
+  std::string path;
+  Result result;
+};
+
+Row& operator<<(Row& row, const SignatureInformation& signature_info) {
+  row["path"] = signature_info.path;
+
+  switch (signature_info.result) {
+  case SignatureInformation::Result::Valid: {
+    row["result"] = "valid";
+    break;
+  }
+
+  case SignatureInformation::Result::Trusted: {
+    row["result"] = "trusted";
+    break;
+  }
+
+  case SignatureInformation::Result::Invalid: {
+    row["result"] = "invalid";
+    break;
+  }
+
+  case SignatureInformation::Result::Missing: {
+    row["result"] = "missing";
+    break;
+  }
+
+  case SignatureInformation::Result::Distrusted: {
+    row["result"] = "distrusted";
+    break;
+  }
+
+  case SignatureInformation::Result::Untrusted: {
+    row["result"] = "untrusted";
+    break;
+  }
+  }
+
+  return row;
+}
+
+Status toUtf16String(std::wstring& output, const std::string& s) {
```

Comment:
We have a `stringToWstring` function already [implemented here](https://github.com/facebook/osquery/blob/master/osquery/core/windows/wmi.cpp#L20) that you could use, but this looks good -- any chance you could take a look at our implementation to see if it could stand some improvements? ;)

Reply:
They look similar! I am now using that one!